packages/logs/src/boot/startLogs.spec.ts

```diff
@@ -43,9 +43,11 @@ describe('logs', () => {
   let interceptor: ReturnType<typeof interceptRequests>
   let requests: Request[]
   let handleLog: ReturnType<typeof startLoggerCollection>['handleLog']
+  let stopLogs: () => void
   let logger: Logger
   let consoleLogSpy: jasmine.Spy
   let displayLogSpy: jasmine.Spy
+  let cleanupTasks: Array<() => void>
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I'm still thinking that a test helper like this could be very *very* helpful:

```js
const cleanupTasks = []

export function registerCleanupTask(task: () => void) {
  cleanupTasks.push(task)
}

afterEach(() => {
  cleanupTasks.splice(0).forEach(task => task())
})
```

(I already suggested it a few years(!) ago, and I clearly remember that you were against it because it did not follow jasmine concepts though 😄)

Reply:
I am not sure to see what you want to do exactly or the debate you refer to 😕
Could you elaborate on what you would like to achieve?

Reply:
Well, imagine you have the code I wrote above in a `core/test/registerCleanupTask.ts`.

Then, you could refactor those patterns:

```ts
describe('....', () => {
  let cleanupTasks: Array<() => void>

  beforeEach(() => {
    cleanupTasks = []
  })

  afterEach(() => {
    cleanupTasks.forEach((task) => task())
  })

  it('....', () => {
    const { stop } = ...
    cleanupTasks.push(stop)
    ...
  })
```

into:

```ts
import { registerCleanupTask } from "../core/test"

describe('....', () => {
  it('....', () => {
    const { stop } = ...
    registerCleanupTask(stop)
    ...
  })
```

It could also be really helpful to use in other helpers, like `mockClock`:

```ts
import { resetNavigationStart } from '../../src/tools/utils/timeUtils'

export type Clock = ReturnType<typeof mockClock>

export function mockClock(date?: Date) {
  jasmine.clock().install()
  ...
  registerCleanupTask(() => {
      jasmine.clock().uninstall()
  })
  return {
    tick: (ms: number) => jasmine.clock().tick(ms),
    setDate: (date: Date) => jasmine.clock().mockDate(date),
  }
}
```

so tests don't have to worry about cleanup, they can just use the helper when they need to:

```ts
describe('...', () => {
  // no need for an afterEach()

  it('...', () => {
    const clock = mockClock()
  })
})
```

Reply:
The @

Reply:
The only concern that I have with this approach is that we would increasing the coupling of our test helpers with jasmine framework.
At least, having it centralised in the `registerCleanupTask` helper seems better than scattering `afterEach` in our helpers.
Let's try that.